packages/auth/amplify_auth_cognito/example/integration_test/device_tracking_test.dart

```diff
@@ -0,0 +1,185 @@
+        autoConfirm: true,
+        verifyAttributes: true,
+        enableMfa: enableMfa,
+      );
+      addTearDown(() => deleteUser(username));
+
+      if (enableMfa) {
+        final code = getOtpCode(username);
+        final signInRes = await Amplify.Auth.signIn(
+          username: username,
+          password: password,
+        );
+        expect(
+          signInRes.nextStep?.signInStep,
+          'CONFIRM_SIGN_IN_WITH_SMS_MFA_CODE',
+        );
+        final confirmSignInRes = await Amplify.Auth.confirmSignIn(
+          confirmationValue: await code,
+        );
+        expect(confirmSignInRes.isSignedIn, isTrue);
+      } else {
+        final res = await Amplify.Auth.signIn(
+          username: username,
+          password: password,
+        );
+        expect(res.isSignedIn, isTrue);
+      }
+    }
+
+    group('Opt-In', () {
+      setUpAll(() async {
+        await configureAuth(
+          config: amplifyEnvironments['device-tracking-opt-in'],
+        );
+      });
+
+      setUp(signIn);
+
+      test('device is not automatically remembered', () async {
+        expect(await getDeviceState(), DeviceState.tracked);
+      });
+
+      test('rememberDevice starts tracking', () async {
+        expect(await getDeviceState(), DeviceState.tracked);
+        await Amplify.Auth.rememberDevice();
+        expect(await getDeviceState(), DeviceState.remembered);
+      });
+
+      test('forgetDevice stops tracking', () async {
+        expect(await getDeviceState(), DeviceState.tracked);
+        await Amplify.Auth.rememberDevice();
+        expect(await getDeviceState(), DeviceState.remembered);
+        await Amplify.Auth.forgetDevice();
+        expect(await getDeviceState(), DeviceState.untracked);
+      });
+
+      test('fetchDevices lists registered devices', () async {
+        expect(await getDeviceState(), DeviceState.tracked);
+        expect(Amplify.Auth.fetchDevices(), completion(isEmpty));
+        await Amplify.Auth.rememberDevice();
+        expect(Amplify.Auth.fetchDevices(), completion(isNotEmpty));
+      });
+    });
+
+    group('Always', () {
+      setUpAll(() async {
+        await configureAuth(
+          config: amplifyEnvironments['device-tracking-always'],
+        );
+      });
+
+      setUp(signIn);
+
+      test('device is automatically remembered', () async {
+        expect(await getDeviceState(), DeviceState.remembered);
+      });
+
+      test('rememberDevice is a no-op when already tracking', () async {
+        expect(Amplify.Auth.rememberDevice(), completes);
+      });
+
+      test('forgetDevice stops tracking', () async {
+        expect(await getDeviceState(), DeviceState.remembered);
+        await Amplify.Auth.forgetDevice();
+        expect(await getDeviceState(), DeviceState.untracked);
+      });
+
+      test('fetchDevices lists registered devices', () async {
+        expect(Amplify.Auth.fetchDevices(), completion(hasLength(1)));
+      });
+    });
+
+    test(
```

Comment:
Starting here, I got a lot `did not complete` errors in the integration tests.

Reply:
Hmm okay I'll check it out